neuraxle/steps/numpy.py

```diff
@@ -37,34 +37,59 @@ def transform_one(self, data_inputs):
         return data_inputs.flatten()
 
 
-class NumpyConcatenateInnerFeatures(NonFittableMixin, BaseStep):
-    def __init__(self):
+class NumpyConcatenateOnCustomAxis(NonFittableMixin, BaseStep):
+    """
+    Numpy concetenation step where the concatenation is performed along the specified custom axis.
+    :param axis: the axis where the concatenation is performed.
+    :return: NumpyConcatenateOnCustomAxis instance which is inherited by base step.
+    """
+
+    def __init__(self, axis):
+        self.axis = axis
         BaseStep.__init__(self)
         NonFittableMixin.__init__(self)
 
     def transform(self, data_inputs):
+        """
+        Apply the concatenation transformation along the specified axis (specified in constructor).
+        :param data_inputs:
+        :return:
+        Numpy array
+        """
         return self._concat(data_inputs)
 
     def transform_one(self, data_inputs):
+        """
+        Apply the concatenation transformation along the specified axis (specified in constructor)
+        :param data_inputs:
+        :return:
+        Numpy array
+        """
         return self._concat(data_inputs)
 
     def _concat(self, data_inputs):
-        return np.concatenate(data_inputs, axis=-1)
+        return np.concatenate(data_inputs, axis=self.axis)
 
+class NumpyConcatenateInnerFeatures(NumpyConcatenateOnCustomAxis):
+    """
+    Numpy concetenation step where the concatenation is performed along `axis=-1`.
+    :param axis: the axis where the concatenation is performed.
+    :return: NumpyConcatenateOnCustomAxis instance which is inherited by base step.
+    """
 
-class NumpyConcatenateOuterBatch(NonFittableMixin, BaseStep):
     def __init__(self):
-        BaseStep.__init__(self)
-        NonFittableMixin.__init__(self)
-
-    def transform(self, data_inputs):
-        return self._concat(data_inputs)
+        # The concatenate is on the inner features so axis = -1.
+        super().__init__(axis=-1)
 
-    def transform_one(self, data_inputs):
-        return self._concat(data_inputs)
+class NumpyConcatenateOuterBatch(NumpyConcatenateOnCustomAxis):
+    """
+    Numpy concetenation step where the concatenation is performed along `axis=0`.
+    :param axis: the axis where the concatenation is performed.
+    :return: NumpyConcatenateOnCustomAxis instance which is inherited by base step.
```

Comment:
I think we must move this comment to the constructor and there should be a comment here without :param nor :return.

Reply:
Perfect, I did not know, where the docstring for the constructor was going. I will move it accordingly.